File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/person/contact-type/contact-type-lookup-model.ts

export interface IContactTypeLookupModel {
  contactTypeID: number;
  name: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ContactTypeLookupModel implements IContactTypeLookupModel {
  contactTypeID: number;
  name: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ContactTypeLookupModel>) {
    if (!!init) {
      Object.assign<ContactTypeLookupModel, Partial<ContactTypeLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Person/ContactType/Queries/GetContactTypes/ContactTypeLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Person;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Purchasing.VendorContact.Queries.GetVendorContacts;
using AdventureWorks.Application.DataEngine.Sales.StoreContact.Queries.GetStoreContacts;

namespace AdventureWorks.Application.DataEngine.Person.ContactType.Queries.GetContactTypes
{
    public partial class ContactTypeLookupModel : IHaveCustomMapping
    {
        public int ContactTypeID { get; set; }
        public string Name { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Person.ContactType, ContactTypeLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ContactTypeLookupModel, BaseContactType>().IgnoreMissingDestinationMembers();
        }
    }
}
